### PR TITLE
Fix repeat pattern merging and acq_element renaming

### PR DESCRIPTION
**Pull request by chellings**
By @antsr 
Tested on S17

### CODE REVIEW

**Review by nathlacroix — APPROVED** [review]
I added one commit which I think would allow to detect unwanted behaviors/bug if the acquisition element name does not end with a segment name. 
@antsr let me know if this is ok for you

File: pycqed/measurement/waveform_control/sequence.py [hunk 1]
```diff
@@ -239,7 +239,7 @@ def merge(sequences, segment_limit=None, merge_repeat_patterns=True):
                         if ch_name in merged_seqs[-1].repeat_patterns:
                             pattern_prev = \
                                 merged_seqs[-1].repeat_patterns[ch_name]
-                            if pattern_prev[1] != pattern[1]:
+                            if pattern_prev[1:] != pattern[1:]:
```
Comment (antsr) [inline pycqed/measurement/waveform_control/sequence.py:242]: Good catch for a potential bug.

File: pycqed/measurement/waveform_control/segment.py [hunk 2]
```diff
@@ -1257,6 +1259,21 @@ def rename(self, new_name):
...
+                new_acq_elements.add(el[:-(len(old_name) + 1)] + '_' \
+                                     + new_name)
+            else:
+                new_acq_elements.add(el)
+                log.warning(f'Acquisition element name: {el} not ending'
+                            f' with "_segmentname": {old_name}. Keeping '
+                            f'current element name when renaming '
+                            f'the segment.')
+        self.acquisition_elements = new_acq_elements
+
```
Comment (antsr) [inline pycqed/measurement/waveform_control/segment.py:1276]: Looks like this should work well in all cases now.

**Review by antsr — APPROVED** [review]
👍

**Pull request merged**